internal/client/sns.go

```diff
@@ -0,0 +1,95 @@
+//go:generate mockgen -package mocks --destination=../mocks/mock_sns.go . SNSClient
+
+package client
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sns"
+	"github.com/base-org/pessimism/internal/core"
+	"github.com/base-org/pessimism/internal/logging"
+
+	"go.uber.org/zap"
+)
+
+// SNSClient ... An interface for SNS clients to implement
+type SNSClient interface {
+	AlertClient
+}
+
+// SNSConfig ... Configuration for SNS client
+type SNSConfig struct {
+	TopicArn string
+	Endpoint string
+}
+
+type snsClient struct {
+	svc      *sns.SNS
+	name     string
+	topicArn string
+}
+
+// NewSNSClient ... Initializer
+func NewSNSClient(cfg *SNSConfig, name string) SNSClient {
+	if cfg.TopicArn == "" {
+		logging.NoContext().Warn("No SNS topic ARN provided")
+	}
+
+	// Initialize a session that the SDK will use to load configuration,
+	// credentials, and region. AWS_REGION, AWS_SECRET_ACCESS_KEY, and AWS_ACCESS_KEY_ID should be set in the
+	// environment's runtime
+	sess, err := session.NewSession(&aws.Config{
+		Endpoint: aws.String(cfg.Endpoint),
```

Comment:
What would happen if this session were to arbitrarily crash? Would the client do a reset or would all subsequent publishing fail to work until the application is reset?

Reply:
Hmm that's a good question. From the [docs](https://docs.aws.amazon.com/sdk-for-go/api/aws/session/) I've read, I haven't been able to find an answer to this. Maybe we should expect that the session could arbitrarily crash just in the case that it does? If this is the case, I can move the session creation to the function that handles publishing the message, but I'm not sure that's the right way to handle this. Open to ideas here

Reply:
Not super pressing but definitely an edge case to lookout for in the wild - lets just add a comment explaining that this could happen

Reply:
[bc97527](https://github.com/base-org/pessimism/pull/200/commits/bc975275398280809302d537705d883a7a973c39)